exchanges/margin/margin_test.go

```diff
@@ -35,128 +29,90 @@ func TestUnmarshalJSON(t *testing.T) {
+	assert.Equalf(t, SpotIsolated, alien.M, "received '%v' expected '%v'", alien.M, SpotIsolated)
 
 	jason = []byte(`{"margin":"hello moto"}`)
 	err = json.Unmarshal(jason, &alien)
-	if !errors.Is(err, ErrInvalidMarginType) {
-		t.Error(err)
-	}
-	if alien.M != Unknown {
-		t.Errorf("received '%v' expected 'isolated'", alien.M)
-	}
+	require.ErrorIs(t, err, ErrInvalidMarginType)
+	assert.Equalf(t, Unknown, alien.M, "received '%v' expected '%v'", alien.M, Unknown)
 }
 
 func TestString(t *testing.T) {
 	t.Parallel()
-	if Unknown.String() != unknownStr {
-		t.Errorf("received '%v' expected '%v'", Unknown.String(), unknownStr)
-	}
-	if Isolated.String() != isolatedStr {
-		t.Errorf("received '%v' expected '%v'", Isolated.String(), isolatedStr)
-	}
-	if Multi.String() != multiStr {
-		t.Errorf("received '%v' expected '%v'", Multi.String(), multiStr)
-	}
-	if Unset.String() != unsetStr {
-		t.Errorf("received '%v' expected '%v'", Unset.String(), unsetStr)
-	}
+	assert.Equal(t, unknownStr, Unknown.String())
+	assert.Equal(t, isolatedStr, Isolated.String())
+	assert.Equal(t, multiStr, Multi.String())
+	assert.Equal(t, unsetStr, Unset.String())
+	assert.Equal(t, spotIsolatedStr, SpotIsolated.String())
+	assert.Equal(t, cashStr, Cash.String())
+	assert.Equal(t, "", Type(30).String())
 }
 
 func TestUpper(t *testing.T) {
 	t.Parallel()
-	if Unknown.Upper() != strings.ToUpper(unknownStr) {
-		t.Errorf("received '%v' expected '%v'", Unknown.String(), strings.ToUpper(unknownStr))
-	}
-	if Isolated.Upper() != strings.ToUpper(isolatedStr) {
-		t.Errorf("received '%v' expected '%v'", Isolated.String(), strings.ToUpper(isolatedStr))
-	}
-	if Multi.Upper() != strings.ToUpper(multiStr) {
-		t.Errorf("received '%v' expected '%v'", Multi.String(), strings.ToUpper(multiStr))
-	}
-	if Unset.Upper() != strings.ToUpper(unsetStr) {
-		t.Errorf("received '%v' expected '%v'", Unset.String(), strings.ToUpper(unsetStr))
-	}
+	assert.Equal(t, Unknown.Upper(), strings.ToUpper(unknownStr))
+	assert.Equal(t, Isolated.Upper(), strings.ToUpper(isolatedStr))
+	assert.Equal(t, Multi.Upper(), strings.ToUpper(multiStr))
+	assert.Equal(t, SpotIsolated.Upper(), strings.ToUpper(spotIsolatedStr))
+	assert.Equal(t, Cash.Upper(), strings.ToUpper(cashStr))
+	assert.Equal(t, Unset.Upper(), strings.ToUpper(unsetStr))
 }
 
 func TestIsValidString(t *testing.T) {
 	t.Parallel()
-	if IsValidString("lol") {
-		t.Fatal("expected 'false', received 'true'")
-	}
-	if !IsValidString("isolated") {
-		t.Fatal("expected 'true', received 'false'")
-	}
-	if !IsValidString("cross") {
-		t.Fatal("expected 'true', received 'false'")
-	}
-	if !IsValidString("multi") {
-		t.Fatal("expected 'true', received 'false'")
-	}
-	if !IsValidString("unset") {
-		t.Fatal("expected 'true', received 'false'")
-	}
-	if IsValidString("") {
-		t.Fatal("expected 'false', received 'true'")
-	}
-	if IsValidString("unknown") {
-		t.Fatal("expected 'false', received 'true'")
-	}
+	require.False(t, IsValidString("lol"))
```

Comment:
```suggestion
	require.False(t, IsValidString("lol"))
```

Consistency mismatch with assert/require for the above tests and these. These can just be assert since no other tests are dependant on prior ones